Sources/SwiftNoise/Components/Cipher.swift

```diff
@@ -23,11 +25,38 @@ public protocol Cipher {
   func rekey(k: Data) throws -> Data
 }
 
+extension Cipher {
+  func rekey(k: Data) throws -> Data {
+    return try self.encrypt(k: k, n: 0xffffffffffffffff, ad: Data(), plaintext: Data(repeating: 0, count: 32))
+  }
+}
+
+// A helper method to convert Nonce (which is a 64-bit unsigned integer) to Data.
+func nonceToDataBig(n: Nonce) -> Data {
+  var be = CFSwapInt64HostToBig(n)
+  let data = Data(bytes: &be, count: MemoryLayout<UInt64>.size)
+
+  let padding = Data(repeating: 0, count: 4)
+  return padding + data
+}
+
+func nonceToDataLittle(n: Nonce) -> Data {
```

Comment:
Maybe this is a bit confusing to call this `nonceToDataLittle` because it does not strictly encode the nonce in little endian. However, since `AESGCM` and `ChaChaPoly` are the only supported ciphers, I think it is fine. Maybe we need a code comment regarding to why is it encoded in this way? Reference: https://noiseprotocol.org/noise.html#the-chachapoly-cipher-functions

Reply:
Added a comment.

I think the call `CFSwapInt64HostToLittle` should encode to little endian? Correct me if I'm wrong...

Reply:
Oh I mean the extra 4 bytes appended. I was expecting that converting `0x012345678abcdef` would result in the first row of the below results, while it results in the second row in reality.
```
EF CD AB 89 67 45 23 01 00 00 00 00 (expected)
00 00 00 00 EF CD AB 89 67 45 23 01 (the current function)
```

Reply:
Anyway, I think the comment is clear enough for now. I'll resolve this and get this merged. :)